samza-core/src/main/java/org/apache/samza/zk/ZkLeaderElector.java

```diff
@@ -81,6 +86,10 @@ private String getHostName() {
     }
   }
 
+  public interface ZkLeaderElectorListener {
```

Comment:
This is needed by the caller to get notifications on the actions happening in the ZkLeaderElector (because it is async).